src/installer/managed/Microsoft.NET.HostModel/Bundle/TargetInfo.cs

```diff
@@ -53,6 +54,12 @@ public bool IsNativeBinary(string filePath)
             return IsLinux ? ElfUtils.IsElfImage(filePath) : IsOSX ? MachOUtils.IsMachOImage(filePath) : PEUtils.IsPEImage(filePath);
         }
 
+        public string GetAppName(string hostName)
+        {
+            Debug.Assert(!IsWindows || Path.GetExtension(hostName).Equals(".exe", StringComparison.InvariantCultureIgnoreCase));
+            return (IsWindows ? Path.GetFileNameWithoutExtension(hostName) : hostName) + ".dll";
```

Comment:
It.looks like any synthetic extension could work. Curious why using dll and not exe?